kibana-reports/server/model/backendModel.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  FORMAT,
+  REPORT_STATE,
+  REPORT_TYPE,
+  TRIGGER_TYPE,
+} from '../routes/utils/constants';
+
+export type BackendReportInstanceType = {
+  id: string;
+  lastUpdatedTimeMs?: number;
+  createdTimeMs?: number;
+  beginTimeMs: number;
+  endTimeMs: number;
+  roles?: Array<string>;
+  status: BACKEND_REPORT_STATE;
+  statusText?: string;
+  inContextDownloadUrlPath?: string;
+  reportDefinitionDetails: {
+    id: string;
+    lastUpdatedTimeMs: number;
+    createdTimeMs: number;
+    roles?: string[];
+    reportDefinition: BackendReportDefinitionType;
+  };
+};
+
+export type BackendReportDefinitionType = {
+  name: string;
+  isEnabled: boolean;
+  source: {
+    description: string;
+    type: BACKEND_REPORT_SOURCE;
+    id: string;
+    origin: string;
+  };
+  format: {
+    duration: string;
+    fileFormat: string;
+    limit?: number;
+    header?: string;
+    footer?: string;
+  };
+  trigger: {
+    triggerType: string; //TODO:
```

Comment:
what's the `TODO` here?

Reply:
removed todo and use BACKEND_TRIGGER_TYPE